### DISCUSSION TITLE
Client libs

### DISCUSSION THREAD

**Discussion by dannysheridan** [bentoml/OpenLLM#94] [Ideas]
OpenLLM serves LLMs over REST API and has a Python and Javascript client. Are there additional client libraries that you'd like to offer developers?

disclaimer: I'm a maintainer at [Fern](https://github.com/fern-api/fern) that auto-generates clients and publishes them to github + registries.

**Comment by aarnphm** [comment]
Hey, Currently we only have Python client. I'm intending to implement TS, Go, Rust and Lua client

**Comment by parano** [comment]
@dannysheridan does Fern work well with OpenAPI specs? OpenLLM already provides OpenAPI json today

**Comment by dannysheridan** [comment]
Yes, Fern reads in OpenAPI and generates clients in TS, Python, Go, and Java. (Rust is on the roadmap)

Can you point me to your OpenAPI spec to check out?

If you'd rather chat synchronously, here's [my cal](https://calendly.com/meetdannysheridan/intro).

**Reply by aarnphm** [reply]
The openapi.yaml is located within the generated bento (`$(bentoml get llm-bento -o path)/api/openapi.yaml`) or at `localhost:3000/docs.json`

**Reply by dannysheridan** [reply]
@aarnphm what's a good language for us to start with?

**Reply by aarnphm** [reply]
Go or Java would be good

**Reply by dannysheridan** [reply]
Sounds good, we'll start with Java. Do you have a timeline for implementing the SDKs you mentioned?

**Reply by aarnphm** [reply]
SDK should be ready if i understand this correctly?

**Reply by dannysheridan** [reply]
> I'm intending to implement TS, Go, Rust and Lua client

Yeah. Wondering if you have a plan for when you'd implement these.

**Reply by aarnphm** [reply]
oh, I don't have any immediate plan atm. I think TS/JS is more urgent. Currently, I'm busy with inference optimization, so would be great if someone can take initiative on this.

Wouldn't mind to hop on a call to discuss about plan for JS/TS client.

**Reply by aarnphm** [reply]
Though with JS/TS, I need to play with React a bit to see what make sense to be included in TS client

**Reply by dannysheridan** [reply]
If TS is more urgent, I think it makes sense to start there. Here's [my availability](https://calendly.com/meetdannysheridan/intro) for a call.
